tests/trainer/test_trainer.py

```diff
@@ -774,6 +774,36 @@ def predict_step(self, batch, *_):
             trainer_fn(model, ckpt_path="best")
 
 
+@pytest.mark.parametrize("fn", ("validate", "test", "predict"))
+def test_best_ckpt_evaluate_raises_error_with_multiple_ckpt_callbacks(tmpdir, fn):
+    """Test that an error is raised if best ckpt callback is used for evaluation configured with multiple
+    checkpoints."""
+
+    class TestModel(BoringModel):
+        def validation_step(self, batch, batch_idx):
+            self.log("foo", batch_idx)
+            self.log("bar", batch_idx + 1)
+            return super().validation_step(batch, batch_idx)
+
+    ckpt_callbacks = [ModelCheckpoint(monitor="foo", save_top_k=1), ModelCheckpoint(monitor="bar", save_top_k=1)]
```

Comment:
they will probably have overlapping filenames, just a remark

Reply:
that is not of any concern related to what we are testing.